src/api/namespaces/neftydrops/routes/mining.ts

```diff
@@ -0,0 +1,151 @@
+import * as express from 'express';
+
+import { NeftyDropsNamespace } from '../index';
+import { HTTPServer } from '../../../server';
+import {FilterDefinition, filterQueryArgs, mergeRequestData} from '../../utils';
+import {buildRangeCondition} from '../../neftydrops/utils';
+import { dateBoundaryParameters, getOpenAPI3Responses, paginationParameters } from '../../../docs';
+import QueryBuilder from '../../../builder';
+
+export function miningEndpoints(core: NeftyDropsNamespace, server: HTTPServer, router: express.Router): any {
+  const sort_collection = {
+    type: 'string',
+    values: [
+      'collection_name', 'sold_wax', 'sold_nefty'
+    ],
+    default: 'collection_name'
+  };
+  const sort_claimer = {
+    type: 'string',
+    values: [
+      'claimer', 'spent_wax', 'spent_nefty'
+    ],
+    default: 'claimer'
+  };
+
+  function miningFilterQueryArgs(sort: any): FilterDefinition {
+    return {
+      before: {type: 'int', min: 1, default: 0},
+      after: {type: 'int', min: 1, default: 0},
+      page: {type: 'int', min: 1, default: 1},
+      limit: {type: 'int', min: 1, max: 100, default: 100},
+      sort: sort,
+      order: {type: 'string', values: ['asc', 'desc'], default: 'desc'}
+    };
+  }
+
+  function buildClaimsQuery(after?: number, before?: number): string {
+    return ` FROM neftydrops_claims
+             WHERE settlement_symbol IS DISTINCT FROM 'NULL'
+                  AND drops_contract = $1
+                  ${buildRangeCondition('"created_at_time"', after, before)}`;
+  }
+  function buildGroupQuery(group_by: string, 
+      sort: string, order: string, limit: number): string {
+    return ` GROUP BY ${group_by}
+             ORDER BY ${sort} ${order}
+             LIMIT ${limit}`;
+  }
+
+  router.get('/v1/mining/collections', server.web.caching(), async (req, res) => {
+    try {
+      const args = filterQueryArgs(req, miningFilterQueryArgs(sort_collection));
+      
+      let queryString = `SELECT collection_name, 
+      SUM(CASE settlement_symbol WHEN 'WAX' THEN final_price ELSE 0 END) AS sold_wax, 
+      SUM(CASE settlement_symbol WHEN 'NEFTY' THEN final_price ELSE 0 END) AS sold_nefty `
```

Comment:
This one should also use the core_amount

Reply:
most of the rows with NEFTY settlement_symbol have NULL values, is it faulty data or when does that happen?
![Screenshot 2021-09-03 at 22 18 45](https://user-images.githubusercontent.com/15836315/132061444-dcf8bba5-0273-4a78-9ced-ae3cfd25f129.png)

Reply:
These are records before the filler was changed. The drops paid with NEFTY will have that data on new claims.